test/index.ts

```diff
@@ -557,23 +557,55 @@ function testNonEmptyArray() {
 }
 
 function testReadonlyKeys() {
-  type T = { readonly a: number; b: string };
-
-  type Actual = ReadonlyKeys<T>;
-
-  type Expected = "a";
-
-  type Test = Assert<IsExact<Actual, Expected>>;
+  type cases = [
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<string>, keyof string>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<number>, keyof number>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<boolean>, keyof boolean>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<bigint>, keyof bigint>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<symbol>, keyof symbol>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<null>, keyof null>>,
+    // @ts-expect-error primitives not allowed
+    Assert<IsExact<ReadonlyKeys<undefined>, keyof undefined>>,
```

Comment:
(optional) `null` and `undefined` will definitely have no keys for first and last supported versions, therefore you can easily replace them with `never`:

```suggestion
    // @ts-expect-error primitives not allowed
    Assert<IsExact<ReadonlyKeys<null>, never>>,
    // @ts-expect-error primitives not allowed
    Assert<IsExact<ReadonlyKeys<undefined>, never>>,
```